### PR TITLE
Use all config options tests Data.Stream.Concurrent

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/1907

File: test/Streamly/Test/Data/Stream/Concurrent.hs [hunk 1]
```diff
@@ -45,6 +45,12 @@ cmp s eq list =
         stream <- run $ Stream.fold Fold.toList  s
         listEquals eq stream list
 
+cmp2 :: ([Int] -> [Int] -> Bool) -> [Int] -> Stream IO Int -> Property
+cmp2 eq list s =
+    monadicIO $ do
+        stream <- run $ Stream.fold Fold.toList  s
+        listEquals eq stream list
```
Comment (adithyaov) [inline test/Streamly/Test/Data/Stream/Concurrent.hs:52]: This is the same as `cmp` with the order of args changed.

File: test/Streamly/Test/Data/Stream/Concurrent.hs [hunk 2]
```diff
@@ -236,50 +242,78 @@ main = hspec
...
-                        `Async.append2` Stream.fromPure 3))
-               in cmp stream sortEq [0, 1, 2, 3]
-
-        prop1 "append balanced"
-            $ let leaf x y = Stream.fromPure x `Async.append2` Stream.fromPure y
-                  leaf11 = leaf 0 1 `Async.append2` leaf 2 (3 :: Int)
-                  leaf12 = leaf 4 5 `Async.append2` leaf 6 7
-                  stream = leaf11 `Async.append2` leaf12
-               in cmp stream sortEq [0, 1, 2, 3, 4, 5, 6,7]
+        asyncSpec $
```
Comment (adithyaov) [inline test/Streamly/Test/Data/Stream/Concurrent.hs:251]: You can remove `mapM all configs` in the test description.

File: test/Streamly/Test/Data/Stream/Concurrent.hs [hunk 3]
```diff
@@ -236,50 +242,78 @@ main = hspec
...
+
+        asyncSpec $
+            let appWith cfg = Async.combineWith cfg Stream.nil (Stream.fromPure 1)
+            in prop1 "mapM all configs append [] [1]"
+                    . cmp2 sortEq [1] . appWith
+
+
+        asyncSpec $
+            let appWith cfg = Async.combineWith
+                        cfg Stream.nil (Stream.fromPure 1)
```
Comment (adithyaov) [inline test/Streamly/Test/Data/Stream/Concurrent.hs:264]: Does this correspond to `cmp (Async.append [Stream.fromPure 1, Stream.nil]) sortEq [1]`?

File: test/Streamly/Test/Data/Stream/Concurrent.hs [hunk 4]
```diff
@@ -236,50 +242,78 @@ main = hspec
...
+            let appWith cfg =
+                    Async.combineWith cfg
+                        (Stream.fromPure 0)
+                        (Async.combineWith cfg
+                            (Stream.fromPure 1)
+                            (Async.combineWith cfg
+                                (Stream.fromPure 2) (Stream.fromPure 3))
+                        )
+            in prop1 "mapM all configs append right associated"
+                    . cmp2 sortEq [0, 1, 2, 3] . appWith
```
Comment (adithyaov) [inline test/Streamly/Test/Data/Stream/Concurrent.hs:303]: `combineWith` isn't encouraged to be used. I wonder why we are so extensively testing it.

File: test/Streamly/Test/Data/Stream/Concurrent.hs [hunk 5]
```diff
@@ -236,50 +242,78 @@ main = hspec
...
+                    Async.combineWith cfg (Stream.fromPure x)
+                        (Stream.fromPure y)
+                leaf11 cfg =
+                    Async.combineWith cfg (leaf 0 1 cfg) $ leaf 2 (3 :: Int) cfg
+                leaf12 cfg =
+                    Async.combineWith cfg (leaf 4 5 cfg) $ leaf 6 7 cfg
+                appWith cfg =
+                    Async.combineWith cfg (leaf11 cfg) (leaf12 cfg)
+            in prop1 "mapM all configs append balanced"
+                    . cmp2 sortEq [0, 1, 2, 3, 4, 5, 6,7] . appWith
```
Comment (adithyaov) [inline test/Streamly/Test/Data/Stream/Concurrent.hs:316]: Should we only use config options in this? What about rest of the tests?

**Review by adithyaov — COMMENTED** [review]
@harendra-kumar Can you also chip in your thoughts here?
Test cases seem to be much harder to read.

File: test/Streamly/Test/Data/Stream/Concurrent.hs [hunk 6]
```diff
@@ -236,50 +242,78 @@ main = hspec
...
+            let appWith cfg = Async.combineWith
+                        cfg (Stream.fromPure 0) (Stream.fromPure 1)
+            in prop1 "mapM all configs append [0] [1]"
+                    . cmp2 sortEq [0, 1] . appWith
+
+        asyncSpec $
+            let appWith cfg =
+                    Async.combineWith cfg
+                        (Async.combineWith cfg (Stream.fromPure 0) Stream.nil)
+                        (Stream.fromPure 1)
```
[the remark this answers is not in the document]
Comment (harendra-kumar) [reply test/Streamly/Test/Data/Stream/Concurrent.hs:278]: Can use `parList`.

**Review by rnjtranjan — COMMENTED** [review]
looks good.

**Pull request merged**